cpp/src/io/orc/reader_impl.cu

```diff
@@ -178,15 +178,28 @@ class metadata {
    * @brief Filters and reads the info of only a selection of stripes
    *
```

Comment:
Add documentation for exception thrown. (either here or at top level API)

Reply:
Added `@throw` to the doxygen section of the top-level reader class in readers.hpp